lib/dradis/plugins/saint/importer.rb

```diff
@@ -0,0 +1,97 @@
+module Dradis::Plugins::Saint
+  class Importer < Dradis::Plugins::Upload::Importer
+    def import(params={})
+      @issues = {}
+      @hosts = {}
+      file_content = File.read(params[:file])
+
+      logger.info {'Parsing SAINT output file...'}
+      doc = Nokogiri::XML( file_content )
+      logger.info{'Done.'}
+
+      doc.xpath('/report').each do |xml_report|
+        logger.info {'Processing report...'}
+
+        # Process <vulnerability> tags
+        xml_report.xpath('./details/vulnerability').each do |vuln|
+          process_vuln_issue(vuln)
+        end
+
+        # Process <host> tags
+        xml_report.xpath('./overview/hosts/host').each do |host|
+          process_host_item(host)
+        end
+
+        # Process <vulnerabilities> tag
+        xml_report.xpath('./overview/vulnerabilities/host_info').each do |xml_host_info|
+          host_name = xml_host_info.xpath('./hostname').first.text
+          xml_host_info.xpath('./vulnerability').each do |evidence|
+            process_evidence(evidence, host_name)
+          end
+        end
+
+        logger.info {'Report processed...'}
+      end
+
+      true
+    end
+
+    private
+    def process_vuln_issue(xml_vuln)
+      # Create Dradis Issue
+      logger.info{ "\t\t => Creating new issue..." }
+      plugin_id = Digest::SHA1.hexdigest(xml_vuln.xpath('./description').first.text) 
+
+      issue_text = template_service.process_template(template: 'vulnerability', data: xml_vuln)
+      issue = content_service.create_issue(text: issue_text, id: plugin_id)
+
+      # Save the issue for later to be linked to evidences
+      @issues[plugin_id] = issue
+    end
+
+    def process_host_item(xml_host)
+      # Create Dradis node
+      host_name = xml_host.xpath('./hostname').first.text || "Unnamed host"
```

Comment:
could we have a case where we don't have a `hostname` but we do have an IP address?

Reply:
@etdsoft based on the given xml, there are no vulnerabilities with no hostname.